### PR TITLE
Add code style guide + guideline for code smells

### PR DESCRIPTION
**Pull request by ddabble**

### CODE REVIEW

**Review by mahoyen — CHANGES_REQUESTED** [review]
Great work! It is nice to have something to look up when you are unsure on how to do things. 

I only have two comments/clarifications.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,1219 @@
...
+        return (
+                other_long_expr
+                + other_long_expr
+        )
+    return [
+        element
+        for inner_list in list_of_lists
+        for element in inner_list
+        if element is not None
+    ]
```
[the remark this answers is not in the document]
Comment (ddabble) [reply CONTRIBUTING.md:222]: This specific section of the style guide is actually only about where to place the *operator* when wrapping, and not whether to place operands on the same or a new line after opening brackets 😅 Other than that, this style guide does not enforce any further details on how to wrap lines with regards to bracket placement 🙂 (Admittedly, I can mention that I purposefully included a variety of (good/acceptable) examples of how one can wrap lines, in this example snippet 🤠 None of them based themselves on the whether the expression was boolean or a list comprehension or whatnot, though 😅)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,1219 @@
...
+
+### For each HTML tag
+
+#### `<link>` or `<script>` tags in a template that's `include`d elsewhere
+These should be moved to the `<head>` tag of the templates that `include` this template,
+so that the files linked are not potentially linked multiple times.
+Furthermore, it's good practice to add a comment in the original template, saying which files should be linked when `include`-ing the template.
+
+#### Links missing `target="_blank"`
+`<a>` tags leading the user away from the "flow" of the current page, should always have its `target` attribute set to `"_blank"`.
```
[the remark this answers is not in the document]
Comment (ddabble) [reply CONTRIBUTING.md:1165]: Yes, nice suggestion! 😄

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mahoyen) [reply CONTRIBUTING.md:222]: ok :)

**Review by mahoyen — APPROVED** [review]
I have updated my own PR after the most recent change :P

**Comment by ddabble** [comment]
Nice! 😄 I'll review it when I have the time; sorry for the long wait! 😬

**Pull request merged**